test/integration/DelayedRecovery.js

```diff
@@ -0,0 +1,388 @@
+  recoveryControllerAddress,
+  recoveryBlockDelay
+) {
+  const txOptions = {
+    from: deployerAddress,
+    gasPrice: ConfigReader.gasPrice,
+    gas: ConfigReader.gas
+  };
+
+  const createUserWalletTxResponse = await userFactoryInstance.createUserWallet(
+    owners,
+    threshold,
+    recoveryOwnerAddress,
+    recoveryControllerAddress,
+    recoveryBlockDelay,
+    [],
+    [],
+    [],
+    txOptions
+  );
+
+  const { returnValues } = createUserWalletTxResponse.events.UserWalletCreated;
+  const userWalletEvent = JSON.parse(JSON.stringify(returnValues));
+
+  const tokenHolderProxyAddress = userWalletEvent._tokenHolderProxy;
+  const gnosisSafeProxyAddress = userWalletEvent._gnosisSafeProxy;
+
+  return {
+    tokenHolderProxyAddress,
+    gnosisSafeProxyAddress
+  };
+}
+
+function instantiateGnosisSafeProxy(gnosisSafeProxyAddress) {
+  const gnosisSafeAbi = abiBinProvider.getABI(GNOSIS_SAFE_CONTRACT_NAME);
+
+  const gnosisSafeProxy = new auxiliaryWeb3.eth.Contract(gnosisSafeAbi, gnosisSafeProxyAddress);
+
+  return gnosisSafeProxy;
+}
+
+function instantiateRecoveryModuleProxy(recoveryModuleAddressProxy) {
+  const recoveryModuleAbi = abiBinProvider.getABI(DELAYED_RECOVERY_MODULE_CONTRACT_NAME);
+
+  const recoveryModule = new auxiliaryWeb3.eth.Contract(recoveryModuleAbi, recoveryModuleAddressProxy);
+
+  return recoveryModule;
+}
+
+function hashRecoveryModuleDomainSeparator(recoveryModuleAddress) {
+  const RECOERY_MODULE_DOMAIN_SEPARATOR_TYPEHASH = auxiliaryWeb3.utils.keccak256(
+    'EIP712Domain(address delayedRecoveryModule)'
+  );
+
+  return auxiliaryWeb3.utils.keccak256(
+    auxiliaryWeb3.eth.abi.encodeParameters(
+      ['bytes32', 'address'],
+      [RECOERY_MODULE_DOMAIN_SEPARATOR_TYPEHASH, recoveryModuleAddress]
+    )
+  );
+}
+
+function hashRecoveryModuleRecoveryStruct(structTypeHash, prevOwner, oldOwner, newOwner) {
+  return auxiliaryWeb3.utils.keccak256(
+    auxiliaryWeb3.eth.abi.encodeParameters(
+      ['bytes32', 'address', 'address', 'address'],
+      [structTypeHash, prevOwner, oldOwner, newOwner]
+    )
+  );
+}
+
+function hashRecoveryModuleRecovery(recoveryModuleAddress, structTypeHash, prevOwner, oldOwner, newOwner) {
+  const recoveryStructHash = hashRecoveryModuleRecoveryStruct(structTypeHash, prevOwner, oldOwner, newOwner);
+
+  const domainSeparatorHash = hashRecoveryModuleDomainSeparator(recoveryModuleAddress);
+
+  return auxiliaryWeb3.utils.soliditySha3(
+    { t: 'bytes1', v: '0x19' },
+    { t: 'bytes1', v: '0x01' },
+    { t: 'bytes32', v: domainSeparatorHash },
+    { t: 'bytes32', v: recoveryStructHash }
+  );
+}
+
+function signRecovery(recoveryModuleAddress, structTypeHash, prevOwner, oldOwner, newOwner, recoveryOwnerPrivateKey) {
+  const recoveryHash = hashRecoveryModuleRecovery(recoveryModuleAddress, structTypeHash, prevOwner, oldOwner, newOwner);
+
+  const signature = EthUtils.ecsign(EthUtils.toBuffer(recoveryHash), EthUtils.toBuffer(recoveryOwnerPrivateKey));
```

Comment:
Signing could have been done with Mosaic.js EIP712Signer which other integration tests are consuming.
https://github.com/OpenSTFoundation/openst.js/blob/develop/test/integration/WalletOperations.js#L200
Improvement for later versions.

Reply:
1. for now ok, but we should avoid having a logic function like `hashRecoveryModuleRecovery` l144 in the test files;
2. if there is not already a good EIP712 js library out there, then I am in favor of splitting the EIP712 signer out of mosaic.js
3. working on the JLP examples we will gives us better clarity on the options for handling the signers outside of mosaic.js or openst.js